pkg/services/sqlstore/migrations/migrations.go

```diff
@@ -53,6 +53,7 @@ func (*OSSMigrations) AddMigration(mg *Migrator) {
 	}
 	ualert.RerunDashAlertMigration(mg)
 	addKVStoreMigrations(mg)
+	ualert.AddDashboardUIDPanelIDMigrations(mg)
```

Comment:
This migration will run once regardless if `ngalert` is enabled of not.
Assuming that `ngalert` is initially disabled: the migration will run without any effect (no alert rules).
Then, when `ngalert` is enabled [the migration of the legacy alerts](https://github.com/grafana/grafana/blob/9f1de095b750dbbfafacd9c77b2ec1c18b878c9b/pkg/services/sqlstore/migrations/ualert/ualert.go#L53) will run but `AddDashboardUIDPanelIDMigrations` won't run because it is marked as executed from before (there is an entry in the `migration_log` table). Therefore, the `dashboard_uid` and `panel_id` columns won't get updated fro the alert rules generated from the migration of the legacy alerts.
I think that we need to set them also during the `AddDashboardUIDPanelIDMigrations`.